src/components/AddDevice/AddDevice.tsx

```diff
@@ -162,6 +166,11 @@ function SignInPage() {
       navigate(`/verify-email?${newSearchParams.toString()}#${hashParams.toString()}`);
     } catch (error: any) {
       console.log(error);
+      const { message } = error;
+      const parsedUrl = new URL(failure_url || success_url || window.location.origin + (basePath ? `/${basePath}` : ''));
+      parsedUrl.searchParams.set('code', error.code);
+      parsedUrl.searchParams.set('reason', message);
+      window.location.replace(parsedUrl.href);
```

Comment:
I see this part of code gets repeated multiple times, is it possible if we pull this into a util function and repeat on other areas?